### PR TITLE
Use Lift instance from text-1.2.4

### PR DESCRIPTION
**Pull request by oliver-batchelor**

### CODE REVIEW

File: src/Instances/TH/Lift.hs [hunk 1]
```diff
@@ -86,8 +86,8 @@ import qualified Data.Set as Set
 import qualified Data.Tree as Tree
 
 -- Text
-import qualified Data.Text as Text
-import qualified Data.Text.Lazy as Text.Lazy
+import qualified Data.Text as Text ()
```
Comment (christiaanb) [inline src/Instances/TH/Lift.hs:89]: Why not

```
#if !MIN_VERSION_text(1,2,4)
import qualified Data.Text as Text
import qualified Data.Text.Lazy as Text.Lazy
#endif
```

Your current PR makes no sense... you can't re-export instances.

Comment (christiaanb) [reply src/Instances/TH/Lift.hs:89]: Apologies... that was phrased way too harsh. Here's what I should have said: I think it's preferable to conditionally (using CPP) add the orphan instances for `Lift Text` in this module. This way, the `th-lift-instances` can also be build with older versions of `text`, add the orphan `Lift` instances for `text < 1.2.4`, and still compile with `text >= 1.2.4`

Comment (christiaanb) [reply src/Instances/TH/Lift.hs:89]: This is the patch I would've suggested instead: https://github.com/christiaanb/th-lift-instances/commit/97ea9cf709f9a72d4df80c8ac56e72df1c695547

**Comment by bennofs** [comment]
Thanks, I applied the suggested patch from @christiaanb

File: src/Instances/TH/Lift.hs (hunk 1 above)
Comment (hvr) [reply src/Instances/TH/Lift.hs:89]: > This way, the th-lift-instances can also be build with older versions of text

btw, note that there's no need for the *same* release of `th-lift-instances` to build with old and new versions of text; The conditionality doesn't need to be handled at the CPP level; with this PR (+ the revisions that were made to Hackage already for previous releases of `th-lift-instances`) the cabal constraint solver will pick the appropriate release of `th-lift-instances` that's compatible with the respective `text` release.

**Pull request closed without merging**